testdata/p4tc_samples_outputs/add_entry_3_example_parser.h

```diff
@@ -65,6 +65,39 @@ REGISTER_TABLE(hdr_md_cpumap, BPF_MAP_TYPE_PERCPU_ARRAY, u32, struct hdr_md, 2)
 BPF_ANNOTATE_KV_PAIR(hdr_md_cpumap, u32, struct hdr_md)
 REGISTER_END()
 
+u32 getPrimitive32(u8 *a, int size) {
+   return  ((a[2]<<16) | (a[1] << 8) | a[0]);
+}
+u64 getPrimitive64(u8 *a, int size) {
+   if(size < 40) {
+       return  ((a[4] << 32) | (a[3] << 24) | (a[2] << 16) | (a[1] << 8) | a[0]);
+   } else {
+       if(size < 48) {
+           return  ((a[5] << 40) | (a[4] << 32) | (a[3] << 24) | (a[2] << 16) | (a[1] << 8) | a[0]);
+       } else {
+           return  ((a[6] << 48) | (a[5] << 40) | (a[4] << 32) | (a[3] << 24) | (a[2] << 16) | (a[1] << 8) | a[0]);
+       }
+   }
+}
```

Comment:
Just to avoid the warnings

```suggestion
u64 getPrimitive64(u8 *a, int size) {
   if(size < 40) {
       return  (((((u64)a[4]) << 32)) | (((u64)a[3]) << 24) | (((u64)a[2]) << 16) | (((u64)a[1]) << 8) | a[0]);
   } else {
       if(size < 48) {
           return  ((((u64)a[5]) << 40) | (((u64)a[4]) << 32) | (((u64)a[3]) << 24) | (((u64)a[2]) << 16) | (((u64)a[1]) << 8) | a[0]);
       } else {
           return  ((((u64)a[6]) << 48) | (((u64)a[5]) << 40) | (((u64)a[4]) << 32) | (((u64)a[3]) << 24) | (((u64)a[2]) << 16) | (((u64)a[1]) << 8) | a[0]);
       }
   }
}
```